### ISSUE TITLE
CA1508 false-positive when using try finally

### ISSUE THREAD

**Issue by TheBigNeo** [dotnet/roslyn-analyzers#6958]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.4 (Latest)

### Describe the bug

CA1508 displays a false-positive warning when a `try finally` is used without a `catch ` block.

If a `catch` block is present, the warning is not displayed.

### Steps To Reproduce

See code snippet at bottom.

### Expected behavior
No warning

### Actual behavior

CA1508 warning inside `finally`

## Additional context

```csharp
#nullable enable
namespace ConsoleApp1;

public class CA1508
{
    public static void ShowError()
    {
        DisposableClass? disposableClass = null;
        try
        {
            disposableClass = new DisposableClass();
        }
        // catch (Exception e)
        // {
        //     Console.Out.WriteLine(e);
        // }
        finally
        {
            disposableClass?.Dispose(); // CA1508: 'disposableClass' is never 'null'. Remove or refactor the condition(s) to avoid dead code.
        }
    }
}

public class DisposableClass : IDisposable
{
    public DisposableClass() => throw new NotImplementedException();
    public void Dispose() => Console.Out.WriteLine("Dispose");
}

```

**Comment by mavasani** [comment]
Verified this repros with .NET 7 SDK, but not with the latest .NET 8 RC1 SDK: https://dotnet.microsoft.com/en-us/download/dotnet/8.0 or the latest Analyzer NuGet package: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/versions/8.0.0-preview1.23170.2
